Lib/test/test_shelve.py

```diff
@@ -399,6 +411,29 @@ def deserializer(data):
         self.assertRaises(shelve.ShelveError, shelve.Shelf, {}, **kwargs)
         self.assertRaises(shelve.ShelveError, shelve.BsdDbShelf, {}, **kwargs)
 
+    def test_custom_serializer_returns_wrong_type_for_key(self):
```

Comment:
This test now duplicates the one when serializer returns None. As the key is never serialized, I don't think we need this one anymore.